fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2202,6 +2203,29 @@ public boolean isIndexDisabled(@Nonnull String indexName) {
         return sanitizeIndexes(getRecordMetaData().getUniversalIndexes(), index -> !isIndexDisabled(index));
     }
 
+    /**
+     * Gets a map from {@link Index} to {@link IndexState} for all the indexes in the metadata.
+     * This method will not perform any queries to the underlying database and instead satisfies the answer based on the
+     * in-memory cache of store state. However, if another operation in a different transaction
+     * happens concurrently that changes the index's state, operations using the same {@link FDBRecordContext}
+     * as this record store will fail to commit due to conflicts.
+     * @return a map of all the index states.
+     */
+    @Nonnull
+    public Map<Index, IndexState> getAllIndexStates() {
+        final RecordStoreState localRecordStoreState = getRecordStoreState();
+        localRecordStoreState.beginRead();
+        try {
+            return getRecordMetaData().getAllIndexes().stream()
+                    .collect(Collectors.toMap(Function.identity(), index -> {
+                        addIndexStateReadConflict(index.getName());
+                        return localRecordStoreState.getState(index);
+                    }));
```

Comment:
In theory, this could add a read conflict range to the entire `RECORD_STORE_STATE` subspace in one go rather than working on each index individually. That would mean that the transaction would conflict if someone, say, updated the meta-data and added a new index that was immediately marked as `WRITE_ONLY` or `DISABLED`, though the transactions should already conflict on the store header (I would think). I guess it would save some JNI hops.